docs/en/Concurrency-Check.md

```diff
@@ -1,3 +1,106 @@
+
+### Base Classes & Interfaces for Concurrency Stamp
+
+#### `IHasConcurrencyStamp` Interface
+
+It is the base interface for **Concurrency Check** and has a just simple property named `ConcurrencyStamp`. 
+
+```csharp
+public interface IHasConcurrencyStamp 
+{
+    public string ConcurrencyStamp { get; set; }
+}
+```
+
+To be able to use **Concurrency Check** your entity class should inherit from the `IHasConcurrencyStamp` interface directly or indirectly.
+
+#### Base Classes
+
+Most of the **Entity Classes** provided by ABP Framework, inherit from the `IHasConcurrencyStamp` interface. You can see the full list of the base classes:
+
+- `AggregateRoot`, `AggregateRoot<TKey>`
+- `CreationAuditedAggregateRoot`, `CreationAuditedAggregateRoot<TKey>`
+- `AuditedAggregateRoot`, `AuditedAggregateRoot<TKey>`
+- `FullAuditedAggregateRoot`, `FullAuditedAggregateRoot<TKey>`
+
+### Getting Started 
+
+This is a core feature and it's used by the ABP Framework. There are no additional packages required. **Optimistic Locking** is applied for ABP modules by default.
+
+#### Usage
+
+You can either inherit your entity from one of [the base classes](#base-classes):
+
+```csharp
+public class Book : FullAuditedAggregateRoot<Guid>
+{
+    //...
+}
+```
+
+or inherit directly from the `IHasConcurrencyStamp` interface:
+
+```csharp
+public class Book : Entity<Guid>, IHasConcurrencyStamp
+{
+    public string ConcurrencyStamp { get; set; }
+        
+    //...
+}
+```
+
+Then you need to inherit your output and update DTO classes from the `IHasConcurrencyStamp` interface:
+
+```csharp
+public class BookDto : EntityDto<Guid>, IHasConcurrencyStamp 
+{
+    //...
+
+    public string ConcurrencyStamp { get; set; }
+}
+
+public class UpdateBookDto : IHasConcurrencyStamp 
+{
+    //...
+
+    public string ConcurrencyStamp { get; set; }
+}
+```
+
+Also, you need to set the **ConcurrencyStamp** to the entity in the **UpdateAsync** method of your application service, for that purpose you can use the `SetConcurrencyStampIfNotNull` method like as below:
+
+```csharp
+public class BookAppService : ApplicationService, IBookAppService 
+{
+    //...
+
+    public virtual async Task<BookDto> UpdateAsync(Guid id, UpdateBookDto input) 
+    {
+        var book = await BookRepository.GetAsync(id);
+
+        book.SetConcurrencyStampIfNotNull(input.ConcurrencyStamp);
+
+        //set other input values to entity ...
+
+        await BookRepository.UpdateAsync(book);
+    }
+}
+```
+
+After that, when multiple users try to change the same record at the same time, concurrency stamp mismatch occurs and `AbpDbConcurrencyException` will be thrown. You can either handle the exception manually or let the ABP Framework handle it for you. 
+
+ABP Framework shows a user-friendly error message like in the image below, if you don't handle the exception manually.
+
```

Comment:
A better ABP error message can be:

Optimistic concurrency check has been failed. The entity you're working on has modified by another user. Please discard your changes and try again.